src/types.cpp

```diff
@@ -0,0 +1,47 @@
+#include "create/types.h"
+
+namespace create {
+  RobotModel::RobotModel(const std::string& name) {
```

Comment:
What if we change the signature to `RobotModel(const RobotModel& model)` instead?  This way we don't need the ugly if-else block below with the error throwing.